sdk/storage/azure-storage-file/apireview/azure/storage/file/file_client.pyi

```diff
@@ -0,0 +1,48 @@
+    def generate_shared_access_signature(self, permission: Optional[Union[FilePermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., policy_id: Optional[str]=..., ip: Optional[str]=..., protocol: Optional[str]=..., cache_control: Optional[str]=..., content_disposition: Optional[str]=..., content_encoding: Optional[str]=..., content_language: Optional[str]=..., content_type: Optional[str]=...) -> str: ...
+    def create_file(self, size: int, content_settings: Optional[ContentSettings]=..., metadata: Optional[Dict[str, str]]=..., file_attributes: Union[str, NTFSAttributes]=..., file_creation_time: Union[str, datetime]=..., file_last_write_time: Union[str, datetime]=..., file_permission: Optional[str]=..., file_permission_key: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def upload_file(self, data: Any, length: Optional[int]=..., metadata: Optional[Dict[str, str]]=..., content_settings: Optional[ContentSettings]=..., validate_content: bool=..., max_connections: Optional[int]=..., file_attributes: Union[str, NTFSAttributes]=..., file_creation_time: Union[str, datetime]=..., file_last_write_time: Union[str, datetime]=..., file_permission: Optional[str]=..., file_permission_key: Optional[str]=..., encoding: str=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def start_copy_from_url(self, source_url: str, metadata: Optional[Dict[str, str]]=..., timeout: Optional[int]=..., **kwargs: Any) -> Any: ...
+    def abort_copy(self, copy_id: Union[str, FileProperties], timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def download_file(self, offset: Optional[int]=..., length: Optional[int]=..., validate_content: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> Iterable[bytes]: ...
+    def delete_file(self, timeout: Optional[int]=..., **kwargs: Optional[Any]) -> None: ...
+    def get_file_properties(self, timeout: Optional[int]=..., **kwargs: Any) -> FileProperties: ...
+    def set_http_headers(self, content_settings: ContentSettings, file_attributes: Union[str, NTFSAttributes]=..., file_creation_time: Union[str, datetime]=..., file_last_write_time: Union[str, datetime]=..., file_permission: Optional[str]=..., file_permission_key: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def set_file_metadata(self, metadata: Optional[Dict[str, Any]]=..., timeout: Optional[int]=..., **kwargs: Optional[Any]) -> Dict[str, Any]: ...
+    def upload_range(self, data: bytes, start_range: int, end_range: int, validate_content: Optional[bool]=..., timeout: Optional[int]=..., encoding: Any=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def upload_range_from_url(self, source_url: str, range_start: int, range_end: int, source_range_start: int, **kwargs: Any) -> Dict[str, Any]: ...
```

Comment:
We need to covert to the offset/length terminology

Reply:
#7812